### DISCUSSION TITLE
Is there a problem accessing a store like this?

### DISCUSSION THREAD

**Discussion by vonkanehoffen** [pmndrs/zustand#2377] [Q&A] [answered]
So following the docs, I've setup a store like:
```typescript
export const useGlobalStore = create<GlobalState>()(
  persist(
    (set) => ({
      numberPlate: '',
      setNumberPlate: (numberPlate) => set({ numberPlate }),
    ...
```
and I access it like:
```typescript
const numberPlate = useGlobalStore((state) => state.numberPlate);
const setNumberPlate = useGlobalStore((state) => state.setNumberPlate);
```

However I now see I can use a shorter form like:
```typescript
const { numberPlate, setNumberPlate } = useGlobalStore();
```

It seems to work ok. I don't see that in the docs though. Does that mean there's some problem with it?

**Comment by dbritto-dev** [comment, marked as answer]
@vonkanehoffen there's no problem with that when you only have the same fields that you are destructuring but when you have more than that you could face some extra re-renders, that's why we always use selectors but you can do something like this:

```ts
const [numberPlate, setNumberPlate] = useGlobalStore(state => [state.numberPlate, state.setNumberPlate]);
```

**Reply by vonkanehoffen** [reply]
Ah nice. Hadn't seen that pattern. Thanks very much! Loving this lib more and more 😁
